widgetssdk/src/main/java/com/glia/widgets/call/CallView.kt

```diff
@@ -382,49 +382,54 @@ internal class CallView(
             negativeButtonClickListener = {
                 dismissAlertDialog()
                 callController?.endEngagementDialogDismissed()
-            },
-            cancelListener = {
+            }
+
+        )
+
+        alertDialog = Dialogs.showAlertDialog(context, theme, DialogType.ReversedOption(payload)) {
+            setOnCancelListener {
                 it.dismiss()
                 callController?.endEngagementDialogDismissed()
-            },
-            isButtonsColorsReversed = true
-        )
+            }
+        }
     }
 
     private fun showAllowScreenSharingNotificationsAndStartSharingDialog() {
         if (alertDialog == null || !alertDialog!!.isShowing) {
```

Comment:
There is still a bit of duplicated code here. Could you maybe extract the if statement into something more general and instead of showAllowScreenSharingNotificationsAndStartSharingDialog we would have something like createAllowSreenSharingAndStartSharingOptions(): DialogPayload.Option that would create only the option?